src/UCTNodePointer.h

```diff
@@ -39,63 +39,64 @@ class UCTNode;
 //  - std::unique_ptr<UCTNode> pointer;
 //  - std::pair<float, std::int16_t> args;
 
-// WARNING : inflate() is not thread-safe and hence has to be protected
-// by an external lock.
+// All methods should be thread-safe except destructor and when
+// the instanced is 'moved from'.
 
 class UCTNodePointer {
 private:
     // the raw storage used here.
-    // if bit 0 is 0, m_data is the actual pointer.
-    // if bit 0 is 1, bit [31:16] is the vertex value, bit [63:32] is the score.
+    // if bit [1:0] is 1, m_data is the actual pointer.
+    // if bit [1:0] is 0, bit [31:16] is the vertex value, bit [63:32] is the score.
+    // if bit [1:0] is other values, it should assert-fail
     // (C-style bit fields and unions are not portable)
-    mutable uint64_t m_data = 1;
+    mutable std::atomic<std::uint64_t> m_data{2};
```

Comment:
Can we make the "2" some INVALID constant or so? Right now the value is repeated throughout the code.